packages/amplify-cli/src/plugin-helpers/scan-plugin-platform.ts

```diff
@@ -38,12 +38,16 @@ export async function scanPluginPlatform(pluginPlatform?: PluginPlatform): Promi
     await sequential(scanUserLocationTasks);
   }
 
-  if (isPackaged) {
-    pluginPlatform!.pluginDirectories.push(constants.PACKAGED_NODE_MODULES);
+  if (!pluginPlatform.pluginDirectories) {
+    pluginPlatform.pluginDirectories = new PluginPlatform().pluginDirectories;
   }
```

Comment:
This might be removing non-null assertion warning, but it's not great way of achieving that. Previous version of the change was better.
I suggest to 
- either, revert to previous version + increment max warning threshold
- or, address the warning in this whole function (this would be preferred).

as for the later if we do this
```
export async function scanPluginPlatform(pluginPlatform?: PluginPlatform): Promise<PluginPlatform> {
  const pluginPlatformLocal = pluginPlatform || readPluginsJsonFile() || new PluginPlatform();

  pluginPlatformLocal.plugins = new PluginCollection();
```
and use `pluginPlatformLocal` below, this will eliminate plenty of those eslint warnings.

Reply:
Fixed it, thanks!